### PR TITLE
Allow providers to respond to 15thnight texts

### LINKED ISSUE #108: Name and Org fields for Users
User model needs a name field and an organization field that can be edited in the user profile.

Shall we also show these as optional fields in the registration form?

@spectralsun

### LINKED ISSUE #118: Text message responses
Allow the provider to respond to the sent text from 15thnight to accept.

### PR DESCRIPTION
**Pull request by justinnoah**
Handling of text message responses from Providers.

Acceptable 'yes's are 'y' and 'yes' (case doesn't matter)
Text message format from provider can be:
- &lt;yes> &lt;alert id> &lt;msg>
- &lt;alert id> &lt;yes> &lt;msg>
- Logging is done to stdout (or where ever print is redirected)
- For messages that are not properly parsed, the texting user gets no reply back
- authenticates message with twilio

fixes #118

### CODE REVIEW

File: _15thnight/api/response.py [hunk 1]
```diff
@@ -63,3 +66,44 @@ def delete_response(uuid):
...
+
+@response_api.route('/response/sms', methods=['GET'])
+def receive_sms_response():
+    """Handle sms alert response from twilio."""
+
+    # Other variables 15thnight may want to keep track of
+    # msg_sid = request.args.get("MessageSid")
+    # sms_sid = request.args.get("SmsSid")
+    # account_sid = request.args.get("AccountSid")
+    # messaging_service_sid = reuest.args.get("MessagingServiceSid")
```
Comment (nobane) [inline _15thnight/api/response.py]: One or multiple of these could be used to verify validity of the response.

File: manage.py [hunk 2]
```diff
@@ -29,42 +29,65 @@ def seed_categories():
 
     food = Category.get_by_name("Food")
     if not food:
-        food = Category("Food", "", 0)
+        food = Category()
+        food.name = "Food",
+        food.description = ""
+        food.sort_order = 0
```
Comment (nobane) [inline manage.py]: Why was this changed? If the model's init needs to be updated to support the right parameters, then it should be.

File: manage.py [hunk 3]
```diff
@@ -29,42 +29,65 @@ def seed_categories():
...
+#        clothing.name = "Clothing"
+#        clothing.description = "Shirts, shoes, and all things."
+#        clothing.sort_order = 0
+#        clothing.save()
+#    clothing_shirts = Service.get_by_name("Shirts")
+#    if not clothing_shirts:
+#        clothing_shirts = Service(
+#            name="Shirts", description="Shirts of all types and sizes",
+#            category=clothing
+#        )
```
Comment (nobane) [inline manage.py]: Committing commented out code is bad. Don't do it.

Comment (justinnoah) [reply manage.py]: Agreed, that was for a work in progress. Absolutely would not be wanting that in the main branch.

File: manage.py (hunk 2 above)
Comment (justinnoah) [reply manage.py]: Ok, Category **init** was removed at some point and this was never updated to reflect that. I will add it back.

**Comment by justinnoah** [comment]
@spectralsun please give this a look over. There may be wording tweaks and possibly even process tweaks that need to happen. Let me know what you find/want to change.

**Comment by justinnoah** [comment]
Also, a quick note, models have a lot of "changes" because I sorted classes by name.

**Comment by justinnoah** [comment]
@spectralsun Now that we have specific needs to respond to, do we have a standardized format for texts or are we going to have fun making a machine interpret human language?

**Comment by justinnoah** [comment]
Now, with the multitude of services/categories, we may instead have a "soft" yes reply, which means a provider can say they are able to help with an alert, but with a text won't be able to pick the service they can help with until they get onto the website.

@spectralsun does that about sum it up correctly?

**Comment by justinnoah** [comment]
@spectralsun besides conflicts, is this to still be on hold for discussions?

**Comment by justinnoah** [comment]
Closing due to the nature of this issue. It needs to be solved in a much different way.
